scripts/preprocess_data.py

```diff
@@ -0,0 +1,192 @@
+import pandas as pd
+import os, sys
+from collections import defaultdict
+
+
+data_infile = sys.argv[1]
+data_outfile = sys.argv[2]
+
+lines = []
+with open(data_infile, 'r') as f1:
+    for line in f1:
+        if not line.startswith(('set MODEper','set MODEx')):
+            lines.append(line)
+            
+with open(data_outfile, 'w') as f2:
+    f2.writelines(lines)
+    
+parsing = False
+
+data_all = []
+data_out = []
+data_inp = []
+
+output_table = []
+
+with open(data_infile, 'r') as f:
+    for line in f:
+        if line.startswith('set YEAR'):
+            start_year = line.split(' ')[3]
+        if line.startswith('set COMMODITY'): # Extracts list of COMMODITIES from data file. Some models use FUEL instead. 
+            fuel_list = line.split(' ')[3:-1]
+        if line.startswith('set FUEL'): # Extracts list of FUELS from data file. Some models use COMMODITIES instead. 
+            fuel_list = line.split(' ')[3:-1]
+        if line.startswith('set TECHNOLOGY'):
+            tech_list = line.split(' ')[3:-1]
+        if line.startswith('set STORAGE'):
+            storage_list = line.split(' ')[3:-1]
+        if line.startswith('set MODE_OF_OPERATION'):
+            mode_list = line.split(' ')[3:-1]
+
+with open(data_infile, 'r') as f:
+    for line in f:
+        if line.startswith(";"):
+            parsing = False   
+        if parsing:
+            if line.startswith('['):
+                fuel = line.split(',')[2]
+                tech = line.split(',')[1]
+            elif line.startswith(start_year):
```

Comment:
I needed to use `line.rstrip(':= ;\n')` to deal with the different line endings on the year lines:
```python                
elif line.startswith(start_year):
    years = line.rstrip(':= ;\n').split(' ')[0:]
    years = [i.strip(':=').strip() for i in years]
```